[prometheus.remote_write] Add troubleshooting steps for ooo errors (#1889)
grafana/alloy: docs/sources/reference/components/prometheus/prometheus.remote_write.md

}
```

## Troubleshooting

### Out of order errors

> You may sometimes see an "out of order" error in {{< param "PRODUCT_NAME" >}}'s logs.

Reviewer: ```suggestion
You may sometimes see an "out of order" error in the {{< param "PRODUCT_NAME" >}} log files.
```

We should try to avoid making Alloy possessive (using `Alloy's`)

Author: I keep making this mistake :) Thanks for correcting me again!

> This means that {{< param "PRODUCT_NAME" >}} sent a metric sample which has an older timestamp than a sample which the database already ingested.

Reviewer: ```suggestion
This means that {{< param "PRODUCT_NAME" >}} sent a metric sample that has an older timestamp than a sample that the database already ingested.
```

The easy way to decide if you should use which or that is if you can remove the part of the sentence that you are writing, it's probably "which", but if you can't remove the part of the sentence, then it's probably "that".

My kitchen, which is in the front of the house, is painted yellow.
vs.
The presentation that you gave yesterday was excellent.

If your database is Mimir, the exact name of the [Mimir error][mimir-ooo-err] is `err-mimir-sample-out-of-order`.

The most common cause for this error is that there is more than one {{< param "PRODUCT_NAME" >}} instance scraping the same target.
To troubleshoot, take the following steps in order:
1. If you use clustering, check if the number of {{< param "PRODUCT_NAME" >}} instances changed at the time the error was logged.

>    This is the only situation in which it is normal to experience an our of order error.

Reviewer: ```suggestion
   This is the only situation in which it is normal to experience an out of order error.
```

>    The error would only happen for a short period of time, until the cluster stabilizes and all {{< param "PRODUCT_NAME" >}} instances have a new list of targets.

Reviewer: ```suggestion
   The error would only happen for a short period, until the cluster stabilizes and all {{< param "PRODUCT_NAME" >}} instances have a new list of targets.
```

   Since the time duration for the cluster to stabilize is expected to be much shorter than the scrape interval, this isn't a real problem.

>    If the out of order error you are seeing is not related to scaling of clustered collectors, it must be investigated.

Reviewer: ```suggestion
   If the out of order error you see is not related to scaling of clustered collectors, it must be investigated.
```

1. Check if there are active {{< param "PRODUCT_NAME" >}} instances which should not be running.

>    There may be an older collector instance which wasn't shut down before a new one was started.

Reviewer: ```suggestion
   There may be an older {{< param "PRODUCT_NAME" >}} instance that wasn't shut down before a new one was started.
```

2. Inspect the configuration to see if there could be multiple {{< param "PRODUCT_NAME" >}} instances which scrape the same target.

> 3. Inspect the WAL to see which {{< param "PRODUCT_NAME" >}} instance sent those metric samples.

Reviewer: This is why we ask everyone to use "lazy numbering" and use `1.` for every number and let the site generator handle the numbering sequence :-)

> 2. Inspect the configuration to see if there could be multiple {{< param "PRODUCT_NAME" >}} instances which scrape the same target.
> 3. Inspect the WAL to see which {{< param "PRODUCT_NAME" >}} instance sent those metric samples.

Reviewer: ```suggestion
1. Inspect the configuration to see if there could be multiple {{< param "PRODUCT_NAME" >}} instances which scrape the same target.
1. Inspect the WAL to see which {{< param "PRODUCT_NAME" >}} instance sent those metric samples.
```

>    The WAL is located in a directory which is set by the `--storage.path` argument of the [run command][run-cmd].

Reviewer: ```suggestion
   The WAL is located in a directory set by the [run command][run-cmd] `--storage.path` argument.
```
